File: mdp_python/worker.py
import sys
from lib import MajordomoWorker

def main():
    verbose = '-v' in sys.argv
    worker = MajordomoWorker('tcp://localhost:5555', 'echo', verbose)

    reply = None
    while True:
        request = worker.recv(reply)
        if request is None:
            break
        reply = request # echo is complex :D

if __name__ == '__main__':
    main()

File: mdp_rust/examples/worker.rs
extern crate majordomo;
extern crate pretty_env_logger;
use majordomo::MajordomoWorker;

fn main() {
    pretty_env_logger::init_timed();
    
    let mut worker = MajordomoWorker::connect("tcp://localhost:5555", "echo", 2000)
        .expect("Couldn't connect to the broker");

    loop {
        let (c_addr, id, payload) = worker.recv().expect("Couldn't receive a request");
        worker.reply(c_addr, id, payload).expect("Couldn't send a reply");
    }
}
